spec.md

```diff
@@ -289,6 +290,34 @@ encapsulated within the `data` attribute.
 * Constraints:
   * OPTIONAL
 
+# Minimum Supported Event Size
+
+In order to increase interoperability, all CloudEvent consumers SHOULD accept
+events that follow these rules:
+
+* The number of attributes does not exceed 100. Each index of a `Map` is
+  considered an attribute.
+* All attribute names are 20 characters or less long.
+* All indexes of `Maps` are 20 characters or less long.
+* All `Binary` attributes do not exceed a size of 1KB.
+* All `String` attributes (including String expressions like `URI-reference` and
+  `Timestamp`) do not exceed a size of 1KB.
+* `Any` attributes containing a `Binary` or a `String` follow the respective
+  rule above. The `data` attribute is excempt from this rule.
+* For a `Map`, the `Any`-typed values follow the rules above.
+* The total size of all attributes (including the `data` attribute) does not
+  exceed 40KB/64KB.
+
+The size of a `String` attribute is measured by encoding it in
+[UTF-8](https://tools.ietf.org/html/rfc3629).
+The size of an `Integer` attribute is 4 bytes.
+
+CloudEvent consumers MAY reject events that do not follow these rules and MAY
+reject messages that are not minified (e.g. contain unnecessary white-space).
```

Comment:
What's the rationale for rejecting messages with additional whitespace that are still within the message length limits (assuming that the consumer is allowed to discard the whitespace)?

Reply:
What is the message length limit? 😉 

This PR does not define a message length limit. It defines limits on some parts of the message. Other parts of the message are not limited.

E.g. for JSON, this PR limits the length of all names and all values, as well as how many names and values can appear in a message. One can use that info to calculate the maximum size of a minified JSON, and then reject all requests that exceed this size. Then one is doing exactly what you ask for: Accept all messages within the maximum message size, even if they contain more whitespace.

So, there is only a rationale for rejecting messages with excessive additional whitespace (discarding 1GB of whitespace from a JSON document is still expensive 😉 ), but without a clearly defined message length limit, I have a hard time drawing a line at which point one is allowed to reject a message. Suggestions welcome.

Reply:
Maybe "MAY reject messages which contain more than 1KB of whitespace" or "MAY reject messages which contain whitespace which extends the total JSON message length beyond 64KB"?

I'm worried that the MAY gives license to drop pretty-printed and otherwise conformant messages, which seems a bit punitive.

Reply:
I like the idea - I've tried to incorporate it without explicitly mentioning JSON. Does it sound good to you?